app/scripts/directives/metrics-date-type-selector.js

```diff
@@ -13,6 +14,31 @@ angular.module('openhimConsoleApp')
         if (!scope.onChange) {
           scope.onChange = function(){};
         }
+
+        scope.customDateBreakdownChange = function() {
+          var from = moment(scope.selected.from);
+          var until = moment(scope.selected.until);
+
+          var diff = function(unit) {
+            return Math.abs(from.diff(until, unit));
+          };
+
+          var maxBreakdown = 120;
+
+          if (from.isAfter(until)) {
+            scope.selected.from = scope.selected.until;
+          } else if (scope.selected.type === 'minute' && diff('minutes') > maxBreakdown) {
+            scope.selected.type = 'hour';
+          } else if (scope.selected.type === 'hour' && diff('hours') > maxBreakdown) {
+            scope.selected.type = 'day';
+          } else if (scope.selected.type === 'day' && diff('days') > maxBreakdown) {
+            scope.selected.type = 'month';
+          } else if (scope.selected.type === 'month' && diff('months') > maxBreakdown) {
+            scope.selected.type = 'year';
+          }
```

Comment:
This only moves the disaggregation up one level, the next level can still be above 120. We need to scan until we find a breakdown that is < 120. Also, it may be better if the relevant options are disabled in the drop box as a visual indicator to the user about what is happening.